includes/class-sensei-course.php

```diff
@@ -2784,22 +2785,47 @@ public static function course_archive_featured_filter( $query ) {
 	}
 
 	/**
-	 * if the course order drop down is changed
+	 * Set the sorting options based on the query parameter and configuration in the Courses archive page.
 	 *
-	 * Hooked into pre_get_posts
+	 * Hooked into pre_get_posts.
 	 *
 	 * @since 1.9.0
 	 * @param WP_Query $query
 	 * @return WP_Query $query
 	 */
-	public static function course_archive_order_by_title( $query ) {
+	public static function course_archive_set_order_by( $query ) {
 
-		if ( isset( $_REQUEST['course-orderby'] ) && 'title' == $_REQUEST['course-orderby']
-			&& 'course' === $query->get( 'post_type' ) && $query->is_main_query() ) {
-			// setup the order by title for this query
-			$query->set( 'orderby', 'title' );
-			$query->set( 'order', 'ASC' );
+		// Applies only to Course archive page.
+		if ( ! $query->is_post_type_archive( 'course' ) ) {
```

Comment:
Tried moving this check to the constructor where the filter is hooked but at that time this function is not working yet.